src/writeData/components/PluginCreateConfigurationFooter.tsx

```diff
@@ -43,10 +48,24 @@ const PluginCreateConfigurationFooterComponent: FC<Props> = props => {
     return Boolean(telegrafConfig)
   }, [telegrafConfig])
 
+  const {contentID} = useParams<ParamsType>()
+
   useEffect(() => {
     if (telegrafConfig) {
       const {config} = telegrafConfig
-      onUpdateTelegraf({...telegrafConfig, config: `${config}${pluginConfig}`})
+      const position =
+        typeof config === 'string'
+          ? config.indexOf(`[[inputs.${contentID}]]`)
+          : -1
+      const updatedConfig =
+        position === -1
+          ? `${config}${pluginConfig}`
+          : `${config.substring(0, position)}${pluginConfig}`
```

Comment:
i think these two expressions should be put in the same `if` statement. if I'm reading this right, if `config === 'string'`, then `updatedConfig` gets set to
```ts
`${config.substring(0, position)}${pluginConfig}`
```

otherwise it's set to

```ts
`${config.substring(0, position)}${pluginConfig}`
```

and `position` is just a transitory variable. I think having these in the same `if` statement would make the intent a little more clear.

if you don't decide to do this, i think at the very least the ternary expressions should be parallel-constructed so that the cases line up.

in the first ternary, if the expression `config === 'string'` is true, `position` gets set to the `indexOf` in the first case, or `-1` in the second case. in the second ternary, they're flipped so that if `-1` is used, it's the first case and the second case maps to getting the `indexOf`

Reply:
I see what you mean about them being confusing. Yes, the second ternary, for updatedConfig, should be constructed differently. I will use `position >= 0` as the comparator instead.

`position` is not transitory. It is needed, because even if `typeof config === 'string` it is possible for `position` to be -1, when the `config.indexOf` call cannot find a match.